### ISSUE TITLE
Feature request: option to unlink axes for rows/columns

### ISSUE THREAD

**Issue by kleinschmidt** [JuliaPlots/AlgebraOfGraphics.jl#203]
Inspired by the ggplot2 [`scales="free[_(x|y)]"` for `facet_grid`](https://ggplot2.tidyverse.org/reference/facet_grid.html#examples).  Sometimes I've `stack`ed a dataframe with multiple variables and want to plot the values on the y-axis, but they're on different scales.  Currently all the y-axes are linked when faceting.  

I'm not sure what the right interface for this would be, given that faceting variables are defined by `mapping`.

Alternatively, I'd appreciate a suggestion about how I could go about doing this un-linking as a post-processing step.

**Comment by piever** [comment]
Somewhat related remark: hopefully stacking variables is less relevant here than in ggplot2 because of [wide data support](http://juliaplots.org/AlgebraOfGraphics.jl/dev/generated/gallery/#Wide-data), which should not link scales by default.

That being said, AoG should definitely support "free scales". What I was thinking is that one subplot (ie, one axis) can have at most one scale for a given attribute. For facet plots, I would propose the following customization. One can pass custom linking strategy via a `links` keyword to `draw`, that is

```julia
draw(plt, links=(x = :col, y = (:row, :col), color = ())
```

to link `x` axes within each column, link all `y` axes and not link `color` at all (each subplot gets a color bar. I imagine `y = :` could be sugar for `y = (:row, :col)`, and `color = nothing` could be sugar for `color = ()` (within the `links` named tuple).

## Things to confirm

1. Only subsets of `(:row, :col)` are sensible things to pass to `links`.
2. If the linking is "too weak", eg `links = (x = (),)`, then the inner ticks are not removed in the facet grid, as there is no guarantee that the values will match.

I think ggplot2 enforces links in case 2., but I don't like the idea of ignoring something the user sets.

Point 1. seems logical, we'd probably just need to make sure that it makes sense also to compute shared attributes in analyses (see https://github.com/JuliaPlots/AlgebraOfGraphics.jl/issues/199). I think it's reasonable that bins in a histogram (or extrema in a kde plot) are either computed globally, or by facet row / col, or per subplot, whereas using different bins within a same subplot seems a bit suspicious.

Does that make sense?

**Comment by tbenst** [comment]
Just came across a use case for this: I have a two-level categorical hierarchy, "major" and "minor", where I want `row=:major` and `x=:minor`. Currently, the minor label is duplicated for all rows

**Comment by kleinschmidt** [comment]
Yeah I think that makes sense @piever! I wonder if `links` is a bit too telegraphic and something more explicit like `scale_links` or `axis_links` would make it clearer what's going on.

**Comment by flyaflya** [comment]
Are you sure wide data scales are unlinked by default?  If so, what am I doing wrong here:

```
using CairoMakie, AlgebraOfGraphics

df = (a = randn(100), b = randn(100) .* 100, draw = Array(1:100))

plt = data(df) * visual(Lines) * 
        mapping(:draw,[:a,:b], 
                col = dims(1), 
                row = dims(2))

draw(plt) ## notice the y-axis is linked 
## can the y-axis scales be free?
```

Here is what I get where the y-axis scales are linked when I do not want them to be (note: I am not sure what `dims()` is doing, but this seems to get me close ... would rather just say `layout = [:a,:b]`:
![image](https://user-images.githubusercontent.com/17184441/133680458-ca63a966-c4d1-41f1-9afc-c0c5f17e18c5.png)

**Comment by piever** [comment]
Fixed by #287
